fiftyone/server/lightning.py

```diff
@@ -317,7 +317,7 @@ async def _do_async_query(
     filter: t.Optional[t.Mapping[str, str]],
 ):
     if isinstance(query, DistinctQuery):
-        if query.has_list and not query.filters:
+        if query.has_list:
             return await _do_distinct_query(collection, query, filter)
```

Comment:
not sure about mongo, but distinct scans in most dbs are really expensive and slow... have you confirmed that relying more on distinct improves performance?

Reply:
A distinct scan with high cardinality is not good, yes. I am not aware of an aggregation pipeline that can return the first N sorted results using a multikey index, though. But we should investigate more